zipline/utils/sentinel.py

```diff
@@ -27,14 +41,14 @@ def __deepcopy__(self, _memo):
         def __copy__(self):
             return self
 
-    cls = type(result)
-    cls.__name__ = name
+    cls = type(Sentinel)
     try:
         # traverse up one frame to find the module where this is defined
-        cls.__module__ = sys._getframe(1).f_globals.get(
-            '__name__',
-            '__main__',
-        )
-    except (AttributeError, ValueError):
-        pass
-    return result
+        cls.__module__ = sys._getframe(1).f_globals['__name__']
+    except (AttributeError, ValueError, KeyError):
```

Comment:
> (basically this would need to be the top frame)

This code is running inside a function though; how could there not be a stack frame above us?  Do we hit that case if this gets called directly from Python embedded in a C application?